portals/publisher/src/main/webapp/source/src/app/components/Shared/PoliciesUI/AttachedPolicyList.tsx

```diff
@@ -42,47 +42,83 @@ interface AttachedPolicyListSharedProps {
     AttachedPolicyCard: any;
 }
 
-const AttachedPolicyListShared: FC<AttachedPolicyListSharedProps> = ({
-    currentPolicyList,
-    setCurrentPolicyList,
-    currentFlow,
-    target,
-    verb,
-    allPolicies,
-    isAPILevelPolicy,
-    sensors,
-    handleDragEnd,
-    policyListToDisplay,
-    AttachedPolicyCard,
-}) => {
-    return (
-        <>
-            <DndContext
-                sensors={sensors}
-                collisionDetection={closestCenter}
-                onDragEnd={handleDragEnd}
-            >
-                <SortableContext
-                    items={currentPolicyList.map((item) => item.uniqueKey)}
-                    strategy={horizontalListSortingStrategy}
+// Option 1: `listOriginatedFromCommonPolicies` and `isApiRevision` are provided
+interface AttachedPolicyListWithCommonProps extends AttachedPolicyListSharedBaseProps {
+    listOriginatedFromCommonPolicies: string[];
+    isApiRevision: boolean;
+}
+
+// Option 2: Neither `listOriginatedFromCommonPolicies` nor `isApiRevision` are provided
+interface AttachedPolicyListWithoutCommonProps extends AttachedPolicyListSharedBaseProps {
+    listOriginatedFromCommonPolicies?: undefined;
+    isApiRevision?: undefined;
+}
+
+// Combine the two using a union type
+type AttachedPolicyListSharedProps = AttachedPolicyListWithCommonProps | AttachedPolicyListWithoutCommonProps;
+
+const AttachedPolicyListShared: FC<AttachedPolicyListSharedProps> = (props) => {
+    if ('listOriginatedFromCommonPolicies' in props) {
+        return (
+            <>
+                <DndContext
+                    sensors={props.sensors}
+                    collisionDetection={closestCenter}
+                    onDragEnd={props.handleDragEnd}
+                >
+                    <SortableContext
+                        items={props.currentPolicyList.map((item) => item.uniqueKey)}
+                        strategy={horizontalListSortingStrategy}
+                    >
+                        {props.policyListToDisplay.map((policy: AttachedPolicy) => (
+                            <props.AttachedPolicyCard
+                                key={policy.uniqueKey}
+                                policyObj={policy}
+                                currentPolicyList={props.currentPolicyList}
+                                setCurrentPolicyList={props.setCurrentPolicyList}
+                                currentFlow={props.currentFlow}
+                                target={props.target}
+                                verb={props.verb}
+                                allPolicies={props.allPolicies}
+                                isAPILevelPolicy={props.isAPILevelPolicy}
+                                listOriginatedFromCommonPolicies={props.listOriginatedFromCommonPolicies}
+                                isApiRevision={props.isApiRevision}
```

Comment:
Shall we remove the prop from these instances?

Reply:
relates to https://github.com/wso2/apim-apps/pull/809#discussion_r1901575381